apps/frontend/src/hooks/common/useTypeSafeSearchParams.ts

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect, useCallback } from 'react';
```

Comment:
question: I do not understand if this hook actually does the type conversion or just marks the type. 
I think searchParams always returns strings.
so would not this result in the following?
```ts
convertToTypedParams({userId: "1"}) => {userId: "1"}:{userId: number}
```
So the type would be number while it is actually a string?